From 686c2210ca79db69272c2e472181799cd9d0f452 Mon Sep 17 00:00:00 2001
From: Torsten Paul <Torsten.Paul@gmx.de>
Date: Sat, 29 Dec 2018 18:22:21 +0100
Subject: [PATCH 1/5] Fix handling of installation dirs using official setup.

---
 CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index afa0461..4ba4a89 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,9 +11,7 @@ set(LIB3MF_VERSION_MAJOR 1)				# increase on every backward-compatibility breaki
 set(LIB3MF_VERSION_MINOR 8)				# increase on every backward compatible change of the API
 set(LIB3MF_VERSION_MICRO 0)				# increase on on every change that does not alter the API
 
-set(CMAKE_INSTALL_BINDIR bin CACHE PATH "directory for installing binary files")
-set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "directory for installing library files")
-set(CMAKE_INSTALL_INCLUDEDIR include/lib3mf CACHE PATH "directory for installing header files")
+include(GNUInstallDirs)
 
 # By default no COM interface is built
 # use the cmake option -DNMR_COM_NATIVE:BOOL=ON to build the COM interface
-- 
2.19.2

